server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.integration;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.arrayContaining;
+import static org.hamcrest.Matchers.is;
+
+import org.apache.james.CassandraJamesServerMain;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.backends.cassandra.EmbeddedCassandra;
+import org.apache.james.mailbox.elasticsearch.EmbeddedElasticSearch;
+import org.apache.james.modules.CassandraJmapServerModule;
+import org.apache.james.webadmin.routes.DomainRoutes;
+import org.apache.james.webadmin.routes.UserRoutes;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.base.Charsets;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+
+public class WebAdminServerIntegrationTest {
+
+    public static final String DOMAIN = "domain";
+    public static final String USERNAME = "username@" + DOMAIN;
+
+    private TemporaryFolder temporaryFolder = new TemporaryFolder();
+    private EmbeddedElasticSearch embeddedElasticSearch = new EmbeddedElasticSearch(temporaryFolder);
+    private EmbeddedCassandra cassandra = EmbeddedCassandra.createStartServer();
+
+    @Rule
+    public RuleChain chain = RuleChain
+        .outerRule(temporaryFolder)
+        .around(embeddedElasticSearch);
+
+    private GuiceJamesServer guiceJamesServer;
+
+    @Before
+    public void setUp() throws Exception {
+        guiceJamesServer = new GuiceJamesServer()
+            .combineWith(CassandraJamesServerMain.cassandraServerModule)
+            .overrideWith(new CassandraJmapServerModule(temporaryFolder, embeddedElasticSearch, cassandra),
+                new WebAdminConfigurationModule());
+        guiceJamesServer.start();
+
+        RestAssured.port = guiceJamesServer.getWebadminPort().orElseThrow(() -> new RuntimeException("Unable to locate Web Admin port"));
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+    }
+
+    @After
+    public void tearDown() {
+        guiceJamesServer.stop();
+    }
+
+    @Test
+    public void postShouldAddTheGivenDomain() throws Exception {
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+        .when()
+            .post(DomainRoutes.DOMAIN + DOMAIN)
+        .then()
+            .statusCode(200);
+
+        assertThat(guiceJamesServer.serverProbe().listDomains()).contains(DOMAIN);
```

Comment:
That's a little strange to not use your api to check your assertions, as you did in you unit tests.
But if you did so you'll have your tests duplicated. So I'm interested by other team's members opinion.